ioeffect/src/main/scala/scalaz/ioeffect/MonadIOInstances.scala

```diff
@@ -0,0 +1,94 @@
+package scalaz.ioeffect
+
+import scalaz._
+import Scalaz._
+
+sealed abstract class MonadIOInstances extends MonadIOInstances1 {
+
+  implicit val taskMonadIO: MonadIO[Task, Throwable] = new MonadIO[Task, Throwable] {
+    override def liftIO[A](io: IO[Throwable, A])(implicit M: Monad[Task]): Task[A] = io
+  }
+}
+
+sealed abstract class MonadIOInstances1 extends MonadIOInstances2 {
+
+  implicit def ioMonadIO[E]: MonadIO[IO[E, ?], E] = new MonadIO[IO[E, ?], E] {
+    override def liftIO[A](io: IO[E, A])(implicit M: Monad[IO[E, ?]]): IO[E, A] = io
+  }
+}
+
+sealed abstract class MonadIOInstances2 {
+  import Isomorphism._
+
+  def fromIso[F[_], G[_], E](D: F <~> G)(implicit E: MonadIO[G, E], M1: Monad[G]): MonadIO[F, E] =
```

Comment:
`fromIso` typically goes on the companion.

If you need access to it from here you can do a self type

`this: MonadIO.type =>`

Reply:
Will just move to companion